Modal responsive improvements

scootyboots opened the pull request:
[//]: # (Note: See https://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)
This is to improve the responsive behavior of the profile modal on smaller/medium size screens. Because the profile component is the same for the profile page and modal, the profile page will also be slightly affected. 

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #14070 #13944

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
- changes only to css and removing a few html elements and unnecessary classes
- mostly accomplished though adding some new media queries 


[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: UUID: 80b23d68-9bd6-4962-8146-636b20aece8c

Commit:
stack profile actions on smaller screens

Commit:
stack avatar and stats for even smaller screens

Commit:
remove unnecessary classes to keep profile nav on the same line

Commit:
adjust media query width

Commit:
refactor stats removing unnecessary classes and simplifying with less elements and relying more on flexbox

Commit:
adjust breakpoints for modal vs pofile page

SabreCat commented:
This is a huge improvement, thank you @scootyboots! I only see one thing worth fixing before merge, on a run through: at minimum width when viewing one's own profile in the modal, the username could stand to have a little bit more margin up top so it doesn't fall underneath the class badge.

![image](https://user-images.githubusercontent.com/4501321/175666237-54be415e-d67e-48db-b4ec-097b23b1e063.png)

Commit:
more margin for avatar

Commit:
handle allocation on middle size more gracefully

scootyboots commented:
Thanks @SabreCat, no problem!

I've added some more margin so the class badge doesn't overlap with the user name. I've also added some more styling for the stats allocation section to handle middle-width screen better.

SabreCat commented:
Thanks for this, @scootyboots, and welcome to the ranks of the Blacksmiths at tier 1! This will go into our staging environment for a few days, and you should see it live on the main site late next week.
